packages/nbmodel/src/ymodel.ts

```diff
@@ -293,6 +293,28 @@ export class YCodeCell
   getOutputs(): Array<nbformat.IOutput> {
     return this.outputs;
   }
+  /**
+   * Create a new YRawCell that can be inserted into a YNotebook
+   */
+  public static create(): YCodeCell {
+    const ymodel = new Y.Map();
+    const ysource = new Y.Text();
+    ymodel.set('source', ysource);
+    ymodel.set('metadata', {});
+    ymodel.set('cell_type', this.prototype.cell_type);
+    ymodel.set('execution_count', this.prototype.execution_count);
```

Comment:
Note that we probably want to implement `execution_count` like this:

```js
get execution_count(): number {
  return this.ymodel.get('execution_count') || 0
}
set execution_count(value: number) {
  this.ymodel.set('execution_count', value)
}
```

So when we call `ymodel.set('execution_count', this.prototype.execution_count);` in a static method, the call to `this.prototype.execution_count` will throw an error because `this.ymodel` doesn't exist on prototype. This is not the same as `cell_type` that never changes.

I think a more elegant implementation for `YCodeCell.create` would be:

```js
public static create(): YCodeCell {
  const cell = super.create()
  cell.nbmodel.set('execution_count', 0) // for some default value
  return cell
}
```

Inheritance on static methods is not often used in practice. But I think that in this case, leveraging static method inheritance is a beautiful solution for the cell factory.

Reply:
Implemented as you have proposed.